lib/config/wrapper.go

```diff
@@ -372,14 +364,17 @@ func (w *Wrapper) Folder(id string) (FolderConfiguration, bool) {
 	defer w.mut.Unlock()
 	for _, folder := range w.cfg.Folders {
 		if folder.ID == id {
-			return folder, true
+			return folder.Copy(), true
 		}
 	}
 	return FolderConfiguration{}, false
 }
 
 // Save writes the configuration to disk, and generates a ConfigSaved event.
 func (w *Wrapper) Save() error {
+	w.mut.Lock()
+	defer w.mut.Unlock()
```

Comment:
Wait, shouldn't this already have been racy before?

Reply:
It should have, but just fixing what I found.

Reply:
Not complaining, just wondering.